Add Daslang language

moleium opened the pull request:
## Description

Adds the Daslang configuration language.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.das
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
            - https://github.com/moleium/linguist/blob/daslang/samples/Daslang/ecs.das (Self-authored for this PR)
            - https://github.com/moleium/linguist/blob/daslang/samples/Daslang/terrain.das (Self-authored for this PR)
        Sample license(s):
            - All samples provided are self-authored for this PR and contributed under the MIT license 
  - [x] I have included a syntax highlighting grammar: https://github.com/moleium/daslang-grammar
  - [x] I have added a color
    - Hex value: `#d3d3d3`
    - Rationale: Close lighter match the language's logo color choice at https://daslang.io
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

lildude commented on vendor/licenses/git_submodule/vscode-dascript.dep.yml:
```diff
@@ -0,0 +1,8 @@
+homepage: https://github.com/GaijinEntertainment/vscode-dascript.git
+license: none
+licenses: []
+notices: []
```
We can not accept unlicensed grammars. The grammar repo needs one of [these licenses](https://github.com/github-linguist/linguist/blob/9b1023ed5d308cb3363a882531dea1e272b59977/vendor/licenses/config.yml#L4-L15) in order for us to accept it.

lildude commented on samples/Daslang/fib.das:
```diff
@@ -0,0 +1,19 @@
+        last = tmp;
+    }
+    return cur;
+}
```
This is not a real-world example and it's also incredibly small so not a very good sample representative of the language from which we can train the classifier. Please replace this with 2-3 real world representative samples.

lildude reviewed (changes requested):
See inline comments

moleium commented:
I've asked the maintainers to license `vscode-dascript`: [issue](https://github.com/GaijinEntertainment/vscode-dascript/issues/12)
Also searching for alternative licensed grammars.

Will update if/when a licensed grammar is secured

moleium commented:
> This is _not_ where you got the sample from 😉 It also doesn't reflect what you've actually included in the PR. Please link to the original source and state the license for every sample file you add.

I've now:
1.  Updated the PR description to list the new `ecs.das` and `terrain.das` samples.
2.  Stated  that these samples are self-authored by me for this PR and are contributed under the MIT license.
3. Changed the syntax highlighting provider link to the updated licensed one.
